packages/stream/index.d.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of the nivo project.
+ *
+ * Copyright 2016-present, Raphaël Benitte.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+import * as React from 'react'
+import {
+    Dimensions,
+    Box,
+    Theme,
+    MotionProps,
+    CartesianMarkerProps,
+    StackOrder,
+    StackOffset,
+    AreaCurve,
+    SvgDefsAndFill,
+} from '@nivo/core'
+import {
+    OrdinalColorsInstruction,
+    CustomColorFunction,
+    InheritedColorProp,
+    InheritedColorFunction,
+} from '@nivo/colors'
+import { LegendProps } from '@nivo/legends'
+import { Scale, ScaleFunc } from '@nivo/scales'
+import { AxisProps } from '@nivo/axes'
+
+declare module '@nivo/stream' {
+    export type TooltipFormatter<T> = (value: T) => React.ReactNode
+    export type TooltipLabel<T> = (value: T) => string
+
+    export type StackFunc<T> = (
+        data: T[]
+    ) => Array<
+        Array<{
+            0: number
+            1: number
+            data: T
+        }>
+    >
+
+    export interface Layer {
+        index: number
+        x: number
+        value: number
+        y1: number
+        y2: number
+    }
+
+    export type AreaGeneratorFunc = (layers: Layer[]) => string
+
+    export interface Datum {
+        color: string
+        index: number
+        key: string
+        value: number
+        x: number
+        y1: number
+        y2: number
+    }
+
+    export type DatumToNumber = (datum: Datum) => number
+
+    interface OptionalStreamProps<T> extends SvgDefsAndFill<T> {
+        stack: StackFunc<T>
+        xScale: Scale
+        yScale: Scale
+
+        order: StackOrder
+        offset: StackOffset
+        curve: AreaCurve
+        areaGenerator: AreaGeneratorFunc
+
+        margin: Box
+
+        axisTop: AxisProps | null
+        axisRight: AxisProps | null
+        axisBottom: AxisProps | null
+        axisLeft: AxisProps | null
+        enableGridX: boolean
+        enableGridY: boolean
+
+        colors: OrdinalColorsInstruction
+        fillOpacity: number
+        getColor: CustomColorFunction
+
+        borderWidth: number
+        borderColor: InheritedColorProp
+        getBorderColor: InheritedColorFunction
+
+        enableDots: boolean
+        renderDot: StreamDotsItem
+        dotPosition: 'start' | 'center' | 'end'
+        dotSize: DatumToNumber | number
+        getDotSize: DatumToNumber
+        dotColor: InheritedColorProp
+        dotBorderWidth: DatumToNumber | number
+        getDotBorderWidth: DatumToNumber
+        dotBorderColor: InheritedColorProp
+
+        isInteractive: boolean
+        tooltipLabel: TooltipLabel<T>
+        getTooltipLabel: TooltipLabel<T>
```

Comment:
This is part of the internal API, computed by the enhancer, this will be removed as soon as we migrate to hooks, it should be removed.